.github/workflows/regression.yml

```diff
@@ -19,17 +19,17 @@ jobs:
         config_set: [BaseMPI, ReverseMPI, ForwardMPI, BaseNoMPI, ReverseNoMPI, ForwardNoMPI]
         include:
           - config_set: BaseMPI
-            flags: '-Denable-pywrapper=true'
+            flags: '-Denable-pywrapper=true --werror'
```

Comment:
I think this is a wonderful idea. Similarly, I would be very interested in seeing us run valgrind or Address Sanitizer (llvm) with our tests too such that they fail on memory issues (but probably not set to fail on leaks yet to start).